app/Services/EntityService.php

```diff
@@ -594,6 +594,31 @@ public function getEnabledEntities(Campaign $campaign, $except = [])
         return $entityTypes;
     }
 
+    /**
+     * @param array $except
+     * @return array
+     */
+    public function getEnabledEntitiesSorted(Campaign $campaign, bool $singular = true, $except = []): array
+    {
+        $entityTypes = [];
+        foreach ($this->entities() as $element => $class) {
+            if (in_array($element, $except)) {
+                continue;
+            }
+            if ($campaign->enabled($element)) {
+                /** @var MiscModel|mixed $misc */
+                $misc = new $class();
+                if ($singular) {
+                    $entityTypes[$this->singular($element)] = Module::singular($misc->entityTypeId(), __('entities.' . $this->singular($element)));
+                } else {
+                    $entityTypes[$this->singular($element)] = Module::plural($misc->entityTypeId(), __('entities.' . $element));
+                }
+            }
+        }
+        asort($entityTypes);
```

Comment:
I think the best way of doing it would be to use Collator and the users locale to determine the correct order, since the method used for attribute templates requires a multi-dimentional array, and we're working with a single dimentional associative one.